bioptim/optimization/optimal_control_program.py

```diff
@@ -712,52 +712,69 @@ def load(file_path: str) -> list:
             out = [ocp, sol]
         return out
 
+    def get_dynamics(self):
+        list_dynamics = []
+        for nlp in self.nlp:
+            list_dynamics.append(nlp.dynamics_type.type.name)
+        return list_dynamics
+
+    def get_ode_solver(self):
+        list_ode = []
+        for nlp in self.nlp:
+            list_ode.append(nlp.ode_solver.rk_integrator.__name__)
+        return list_ode
+
     def structure_graph(
         self,
-        n_shooting: Union[int, tuple],
         print_to_terminal: bool = False,
+        draw_graph: bool = False,
     ):
 
-        if isinstance(n_shooting, tuple):
-            n_nodes = sum(n_shooting) + 1
-        else:
-            n_nodes = n_shooting + 1
-
-        init_dict = [["Phase", []], ["Mayer", []], ["Lagrange", []], ["Constraints", []]]
-        init_dict = dict(init_dict)
-        list_nodes = [init_dict] * n_nodes
-        for i in range(n_nodes):
-            list_nodes[i] = deepcopy(init_dict)
+        def merge_dicts(a, b):
+            m = a.copy()
+            m.update(b)
+            return m
+
+        def print_terminal(print_terminal: bool, l_dynamics: list, l_ode: list, l_nodes: list, n_phase: int):
+            if print_terminal is True:
+                for phase_idx in range(n_phase):
+                    node_idx = 0
+                    print(f"**********")
+                    print(f"PHASE {phase_idx}")
+                    print(f"DYNAMICS: {l_dynamics[phase_idx]}")
+                    print(f"ODE: {l_ode[phase_idx]}")
+                    print("")
+                    for node_dict in l_nodes[phase_idx]:
+                        print(f"NODE {node_idx}")
+                        print(f"Objectives: ")
+                        print(f"*** Mayer: {node_dict['Mayer']}")
+                        print(f"*** Lagrange: {node_dict['Lagrange']}")
+                        print(f"Constraints: {node_dict['Constraints']}")
+                        print("")
+                        node_idx = node_idx + 1
+
+        # def draw_graph(draw: bool, l_dynamics: list, l_ode: list, l_nodes: list, n_phase: int):
+        #     g = Digraph('graph_test', filename='graph_test.gv')
+        #
+        #     for phase_idx in range(n_phase):
+        #         with g.subgraph(name=f"nlp_{phase_idx}") as c:
+        #             c.attr(style='filled', color='lightgrey')
+        #             c.node_attr.update(style='filled', color='white')
```

Comment:
The import for this is probably at the top of the file. Move it here. I don't think we should prevent from being able to use bioptim when graphviz is not install, but just when someone actually wants to graphs something. Maybe to have something cleaner, this snippet here could be moved in a separate file stored in the gui folder?

Reply:
Done